### PR TITLE
EPMEDU-772 Open Home page if user already logged in

### PR DESCRIPTION
**Pull request by n3gbx**
- Added logic to Splash screen to check whether user profile info saved or not
-- if saved, navigate to Home screen
-- if not, navigate to onboarding
- Added action on logout: delete saved user profile info, clear back stack and navigate to onboarding
- Added SharedFlow event to StateViewModel (thanks @egorikftp)

[video](https://user-images.githubusercontent.com/17089577/184096293-10b0ad89-8831-42f2-ad28-499e5f6edf2c.mp4)

### CODE REVIEW

File: feature/splash/src/main/java/com/epmedu/animeal/splash/presentation/SplashViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+class SplashViewModel @Inject constructor(
+    private val profileRepository: ProfileRepository
+) : ViewModel() {
+
+    private val _event = MutableSharedFlow<Event>()
+    val event: SharedFlow<Event> get() = _event.asSharedFlow()
+
+    init {
+        viewModelScope.launch {
+            delay(2000)
```
Comment (egorikftp) [inline feature/splash/src/main/java/com/epmedu/animeal/splash/presentation/SplashViewModel.kt]: do we need delay?

File: feature/more/src/main/java/com/epmedu/animeal/more/MoreViewModel.kt [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
+import kotlinx.coroutines.launch
+import javax.inject.Inject
 
-internal class MoreViewModel : ViewModel() {
+@HiltViewModel
+internal class MoreViewModel @Inject constructor(
+    private val profileRepository: ProfileRepository
+) : ViewModel() {
+
+    private val _event = MutableSharedFlow<Event>()
```
Comment (egorikftp) [inline feature/more/src/main/java/com/epmedu/animeal/more/MoreViewModel.kt]: What do you think if we add `MutableSharedFlow` into `StateViewModel` for handling events?

cc: @Fileek 

Will be something like this

image

image

Comment (n3gbx) [reply feature/more/src/main/java/com/epmedu/animeal/more/MoreViewModel.kt]: Updated

File: feature/splash/src/main/java/com/epmedu/animeal/splash/presentation/SplashViewModel.kt (hunk 1 above)
Comment (n3gbx) [reply feature/splash/src/main/java/com/epmedu/animeal/splash/presentation/SplashViewModel.kt]: Updated

**Review by Fileek — APPROVED** [review]

File: feature/login/src/main/java/com/epmedu/animeal/login/code/presentation/EnterCodeScreen.kt [hunk 3]
```diff
@@ -36,8 +35,8 @@ internal fun EnterCodeScreen() {
     LaunchedEffect(Unit) {
         focusRequester.requestFocus()
 
-        viewModel.isCodeCorrect.collectLatest { isCorrect ->
-            if (isCorrect) {
+        viewModel.event.collect {
+            if (it is EnterCodeViewModel.Event.NavigateToFinishProfile) {
```
Comment (egorikftp) [inline feature/login/src/main/java/com/epmedu/animeal/login/code/presentation/EnterCodeScreen.kt]: Could you please make import for `EnterCodeViewModel` 

will be
```kotlin
if (it is Event.NavigateToFinishProfile) {
```

File: feature/more/src/main/java/com/epmedu/animeal/more/root/MoreScreen.kt [hunk 4]
```diff
@@ -1,16 +1,30 @@
...
 
 @Composable
 internal fun MoreScreen() {
-    val viewModel: MoreViewModel = viewModel()
+    val viewModel: MoreViewModel = hiltViewModel()
     val navigator = LocalNavigator.currentOrThrow
 
+    LaunchedEffect(Unit) {
+        viewModel.event.collect {
+            if (it is MoreViewModel.Event.NavigateToOnboarding) {
```
Comment (egorikftp) [inline feature/more/src/main/java/com/epmedu/animeal/more/root/MoreScreen.kt]: The same for `MoreViewModel`

**Review by egorikftp — APPROVED** [review]

File: feature/login/src/main/java/com/epmedu/animeal/login/code/presentation/EnterCodeScreen.kt (hunk 3 above)
Comment (n3gbx) [reply feature/login/src/main/java/com/epmedu/animeal/login/code/presentation/EnterCodeScreen.kt]: Updated

File: feature/more/src/main/java/com/epmedu/animeal/more/root/MoreScreen.kt (hunk 4 above)
Comment (n3gbx) [reply feature/more/src/main/java/com/epmedu/animeal/more/root/MoreScreen.kt]: Updated

**Review by Fileek — APPROVED** [review]

**Pull request merged**